### ISSUE TITLE
Svg to Png, gradation is disappear.

### ISSUE THREAD

**Issue by kkeolmusae** [lovell/sharp#3694]
### Is this a possible bug in a feature of sharp, unrelated to installation?

- [X] Running `npm install sharp` completes without error.
- [X] Running `node -e "require('sharp')"` completes without error.

If you cannot confirm both of these, please open an [installation issue](https://github.com/lovell/sharp/issues/new?labels=installation&template=installation.md) instead.

### Are you using the latest version of sharp?

- [X] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

If you cannot confirm this, please upgrade to the latest version and try again before opening an issue.

If you are using another package which depends on a version of `sharp` that is not the latest, please open an issue against that package instead.

### What is the output of running `npx envinfo --binaries --system --npmPackages=sharp --npmGlobalPackages=sharp`?

```

  System:
    OS: macOS 13.4
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 1.75 GB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/Codes/private-project/node_modules/.bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  npmPackages:
    sharp: ^0.32.1 => 0.32.1 

```

### What are the steps to reproduce?

first i install svg image from here.
https://storage.googleapis.com/nftimagebucket/tokens/0xc36442b4a4522e871399cd717abdd847ab11fe88/520923.svg

![image](https://github.com/lovell/sharp/assets/35340846/054f74cc-8a57-4756-a860-b8d24c6f8f90)

and i convert this svg to png.
result: 
<img width="308" alt="image" src="https://github.com/lovell/sharp/assets/35340846/65a17b56-15df-436f-bd8c-f854183642fe">

### What is the expected behaviour?

i expected same color image,
![image](https://github.com/lovell/sharp/assets/35340846/bfb4f7ac-44ac-48a5-a438-7d05dbd66f2e)

but i got black.

### Please provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem

```
const sharp = require("sharp");

(async () => {
  const path = "./nft_uri/520923.svg";

  const file = loadFile(path);
  const metadata = await sharp(Buffer.from(file)).metadata();
  const png = await sharp(Buffer.from(file)).png().toBuffer();
  console.log(png.toString("base64"));
})();
```

### Please provide sample image(s) that help explain this problem
i want colorful image, but i got black and white image. 
<img width="308" alt="image" src="https://github.com/lovell/sharp/assets/35340846/65a17b56-15df-436f-bd8c-f854183642fe">

**Comment by lovell** [comment]
Please see #1378 #2844 #3557

**Comment by kkeolmusae** [comment]
Are you saying that you don't support "svg with embedded base64 encoding svg"?

**Comment by kkeolmusae** [comment]
i did.
```
> brew install librsvg

git clone this repository. (https://github.com/lovell/package-libvips-darwin/tree/master)

> package-libvips-darwin % ./package.sh
./package.sh: line 10: pkg-config: command not found
rm: include/gettext-po.h: No such file or directory
chmod: lib/*.dylib: No such file or directory
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 71: pkg-config: command not found
./package.sh: line 79: pkg-config: command not found
./package.sh: line 81: advdef: command not found
5b9058172a82bb849a23eb58922b88ea8c575baa  libvips--darwin-x64.tar.gz
```

what should i do?  i can't understand this (https://github.com/lovell/sharp-libvips/issues/9)
```
The [librsvg](https://github.com/Homebrew/homebrew-core/blob/master/Formula/librsvg.rb) formula appears to have already shaved the loader.cache yak. Need to remove --without-modules from [here](https://github.com/lovell/package-libvips-darwin/blob/master/.travis.yml#L8) too.
```

**Comment by kkeolmusae** [comment]
i install pkg-config and retry

```
package-libvips-darwin % ./package.sh 
cp: /usr/include/: No such file or directory
chmod: lib/*.dylib: No such file or directory
./package.sh: line 81: advdef: command not found
4fd07723c5a5bf939a46456c6975b2ee24edb268  libvips-8.14.2-darwin-x64.tar.gz
```

how do i use this `libvips-8.14.2-darwin-x64.tar.gz` ?

**Comment by lovell** [comment]
Please don't use anything from the https://github.com/lovell/package-libvips-darwin/ repo; it is archived, outdated and unmaintained.

To globally install libvips on macOS, you should be able to use `brew install vips` then remove and re-install sharp.

**Comment by kkeolmusae** [comment]
aha Thanks!

i try this
```
brew install vips (try reinstall)

npm remove sharp

npm install remove
```

and i retry svg to png using sharp

```
...

const png = await sharp(Buffer.from(file)).png().toBuffer();
console.log(png.toString("base64"));
```

but result is same and nothing change.....OMG......

Am I missing something? Thanks!

**Comment by kkeolmusae** [comment]
I tried various attempts, but I couldn't handle the nested SVG within the SVG. Since the SVG file size is not large (less than approximately 10kb), I decided to use the original SVG file as it is. 

(The reason for wanting to convert the SVG file to PNG or resize it was to reduce its size and convert it into a thumbnail before storing it in the database)

 If anyone has experienced the same issue but found a solution, I hope they can share the solution for the benefit of others.

Thanks to @lovell for their valuable assistance. With that, I will now close this.
